src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -9101,12 +9119,29 @@ private ForStatementSyntax ParseForStatement(SyntaxList<AttributeListSyntax> att
 
                 if (isDeclaration)
                 {
+                    SyntaxToken scopedKeyword = null;
+
+                    if (haveScopedKeyword)
+                    {
+                        scopedKeyword = EatContextualToken(SyntaxKind.ScopedKeyword);
+                    }
+
                     decl = ParseVariableDeclaration();
-                    if (decl.Type.Kind == SyntaxKind.RefType)
+
+                    var declType = decl.Type;
+                    if (declType.Kind == SyntaxKind.RefType)
+                    {
+                        declType = CheckFeatureAvailability(declType, MessageID.IDS_FeatureRefFor);
+                    }
+
+                    if (scopedKeyword != null)
+                    {
+                        declType = _syntaxFactory.ScopedType(scopedKeyword, declType);
+                    }
+
+                    if (declType != decl.Type)
                     {
-                        decl = decl.Update(
-                            CheckFeatureAvailability(decl.Type, MessageID.IDS_FeatureRefFor),
-                            decl.Variables);
+                        decl = decl.Update(declType, decl.Variables);
```

Comment:
nit, could just do the update no matter what.  it will already test that the type actually changed before updating.